stl/inc/memory

```diff
@@ -1504,6 +1556,238 @@ private:
     }
 };
 
+#ifdef __cpp_concepts
+template <class _Ty>
+constexpr _Ty _Div_ceil(_Ty _Val1, _Ty _Val2) noexcept {
+    return (_Val1 % _Val2 == 0) ? (_Val1 / _Val2) : (_Val1 / _Val2 + 1);
+}
+
+template <class _NoThrowIt>
+struct _Uninitialized_rev_destroying_backout { // struct to undo partially constructed ranges in _Uninitialized_xxx
+                                               // algorithms
+    _NoThrowIt _First;
+    _NoThrowIt _Last;
+
+    explicit _Uninitialized_rev_destroying_backout(_NoThrowIt _Dest) : _First(_Dest), _Last(_Dest) {}
+
+    _Uninitialized_rev_destroying_backout(_NoThrowIt _First_, _NoThrowIt _Last_) : _First(_First_), _Last(_Last_) {}
+
+    _Uninitialized_rev_destroying_backout(const _Uninitialized_rev_destroying_backout&) = delete;
+    _Uninitialized_rev_destroying_backout& operator=(const _Uninitialized_rev_destroying_backout&) = delete;
+
+    ~_Uninitialized_rev_destroying_backout() {
+        if (_Last != _First) {
+            do {
+                --_Last;
+                _STD destroy_at(_Last);
+            } while (_Last != _First);
+        }
+    }

```

Comment:
It there a reason you do not simply use a while loop?

```suggestion
        while (_Last != _First) {
            --_Last;
            _STD destroy_at(_Last);
        }
    }

Reply:
I must've been really tired while writing this code. It used do-while because firstly I didn't think of zero-sized arrays and then added "support" for them the laziest way possible...🤣